ztp/policygenerator-kustomize-plugin/Makefile

```
@@ -0,0 +1,29 @@
+# Set kustomize env variable
+export XDG_CONFIG_HOME=./
+
+POLICYGEN_KUSTOMIZE_DIR=./kustomize/plugin/policyGenerator/v1/policygenerator
+KUSTOMIZE_DIR=/tmp
+KUSTOMIZE_BIN=$(KUSTOMIZE_DIR)/kustomize
+KUSTOMIZE_VERSION=4.4.0
+KUSTOMIZE := $(if $(shell command -v kustomize 2>/dev/null), $(shell command -v kustomize), $(KUSTOMIZE_BIN))
+POLICYGEN_DIR := ../policygenerator
+
+.PHONY: build clean
+
+build:
+	@echo "ZTP: Build policy generator binary"
+	$(MAKE) -C $(POLICYGEN_DIR) build
+	mkdir -p $(POLICYGEN_KUSTOMIZE_DIR)
+	cp -r $(POLICYGEN_DIR)/resources $(POLICYGEN_KUSTOMIZE_DIR)/
+	cp $(POLICYGEN_DIR)/policygenerator $(POLICYGEN_KUSTOMIZE_DIR)/PolicyGenerator
+	@if [[ $(KUSTOMIZE) == $(KUSTOMIZE_BIN) ]] && [ ! -f $(KUSTOMIZE) ]; then \
+		echo "kustomize not installed getting kustomize v"$(KUSTOMIZE_VERSION) \
+		&& cd $(KUSTOMIZE_DIR) && curl -m 600 -s "https://raw.githubusercontent.com/kubernetes-sigs/kustomize/master/hack/install_kustomize.sh" | bash -s $(KUSTOMIZE_VERSION); \
+	fi
+	$(KUSTOMIZE) build --enable-alpha-plugins ./
```

Can we split the targets a bit, maybe like this?
```suggestion
.PHONY: all build test clean

all: build test

build:
	@echo "ZTP: Build policy generator kustomize plugin"
	$(MAKE) -C $(POLICYGEN_DIR) build
	mkdir -p $(POLICYGEN_KUSTOMIZE_DIR)
	cp -r $(POLICYGEN_DIR)/resources $(POLICYGEN_KUSTOMIZE_DIR)/
	cp $(POLICYGEN_DIR)/policygenerator $(POLICYGEN_KUSTOMIZE_DIR)/PolicyGenerator

test: build
	@echo "ZTP: Test policy generator kustomize plugin"
	@if [[ $(KUSTOMIZE) == $(KUSTOMIZE_BIN) ]] && [ ! -f $(KUSTOMIZE) ]; then \
		echo "kustomize not installed getting kustomize v"$(KUSTOMIZE_VERSION) \
		&& cd $(KUSTOMIZE_DIR) && curl -m 600 -s "https://raw.githubusercontent.com/kubernetes-sigs/kustomize/master/hack/install_kustomize.sh" | bash -s $(KUSTOMIZE_VERSION); \
	fi
	$(KUSTOMIZE) build --enable-alpha-plugins ./
```

---

okay, but will not add **all** target as it does what **test**  target do, it just build twice

---

We could add actual dependencies so that doesn't happen...  But sure.